opentreemap/treemap/audit.py

```diff
@@ -156,10 +167,36 @@ def add_default_permissions(instance, roles=None, models=None):
         models = leaf_models_of_class(Authorizable) | {MapFeaturePhoto}
 
     for role in roles:
-        _add_default_permissions(models, role, instance)
+        _add_default_field_permissions(models, role, instance)
+
+    _add_default_add_and_delete_permissions(models, roles)
+
+
+def _add_default_add_and_delete_permissions(models, roles):
+    # Each of the 'models' has built-in "add" and "delete" permissions
+    # (from django.contrib.auth, e.g. Plot has "add_plot" and "delete_plot").
+    # Bulk create those permissions for all 'roles' whose default permission
+    # level allows write.
+    role_ids = [
+        role.id for role in roles
+        if role.default_permission_level > FieldPermission.READ_ONLY]
 
+    ThroughModel = Role.instance_permissions.through
+    existing = ThroughModel.objects.filter(role_id__in=role_ids)
+    existing_pairs = [(e.role_id, e.permission_id) for e in existing]
 
-def _add_default_permissions(models, role, instance):
+    perm_names = [Role.permission_codename(Model, action)
+                  for Model in models for action in ['add', 'delete']]
+    perms = Permission.objects.filter(codename__in=perm_names)
+
+    role_perms = [ThroughModel(role_id=role_id, permission_id=perm.id)
+                  for role_id in role_ids for perm in perms
+                  if (role_id, perm.id) not in existing_pairs]
+
+    ThroughModel.objects.bulk_create(role_perms)
+
+
+def _add_default_field_permissions(models, role, instance):
```

Comment:
Passing the instance saves a DB lookup. A vanilla load of a role from the db just has the instance ID, and dereferencing `role.instance` causes a DB lookup. 

(When loading the role I could specify that the instance be pre-loaded, but that seems like overkill since we have the instance right there.)